solutions/usaco-815.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: usaco-815
+source: USACO Gold 2018 Febuary
+title: Taming the Herd
+author: Shijie Ren
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_gold_feb18.html)
+
+## Explanation
+
+Let $\texttt{dp}[i][j]$ be the minimum answer of first $j$ days with $i$ breakouts.
+
+$j$-th day , if log is tampered and cows would NOT breakout:
+
+$$
+\texttt{dp}[i][j]=\texttt{dp}[i][j-1]+1
+$$
+
+$j$-th day , if log is tampered and cows would breakout :
+
+(So $a[j] \not = 0$)
+
+$$
+\texttt{dp}[i][j]=\texttt{dp}[i-1][j-1]+1
+$$
+
+$j$-th day , if log is NOT tampered:
+
+Then $i$-th breakout will be on $(j-a[j])$-th day.
+
+$$\texttt{range\_ans}[l][r]$$ represents minimum cost to replace range $a[l]...a[r]$ with $0...(r-l)$ respectively.
+
+$$
+\texttt{dp}[i][j] = \texttt{dp}[i-1][j-a[j]-1]+\texttt{range\_ans}[j-a[j]][j]);
+$$
+
+Find the maximum among the three.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <cstdio>
+#include <cstring>
+#include <algorithm>
+using namespace std;
+int a[105],range_ans[105][105];
+int dp[105][105];
```

Comment:
1. put 100 in a constant
2. declarations each on their own line
3. (optional) just use vectors

Reply:
if you're gonna reject the optional one could you give a reason?

Reply:
I'll use it now